create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusHandler.java

```diff
@@ -86,6 +95,45 @@ private Reference generateReference(DocTp docTp) {
         return reference;
     }
 
+    private List<TitletextTp> getTitleText(DocTp docTp) {
+        return docTp.getItem().getItem().getBibrecord().getHead().getCitationTitle().getTitletext();
+    }
+
+    private String extractMainTitle(DocTp docTp) {
+        StringBuilder mainTitleBuilder = new StringBuilder();
+        getTitleText(docTp)
+            .stream()
+            .filter(titleTextTp -> titleTextTp.getOriginal().equals(YesnoAtt.Y))
+            .findFirst()
+            .ifPresent(part -> mainTitleBuilder.append(getContentWithXmlTags(part.getContent())));
+        return mainTitleBuilder.toString();
```

Comment:
It is a bit hidden what happens when there is no original title. You could avoid it like this (but better to use the SingletonCollector)
```
    private static final String EMPTY_STIRING = "";

private String extractMainTitle(DocTp docTp) {
        return getTitleText(docTp)
            .stream()
            .filter(this::isOriginalTitle)
            .findAny()
            .map(RichstringWithMMLType::getContent)
            .map(this::getContentWithXmlTags)
            .orElse(EMPTY_STIRING);
    }
```

Reply:
There should be fresher changes than this. (amongst other this should have been fixed)